packages/core/src/index.ts

```diff
@@ -313,38 +313,35 @@ export function buildHooksSystem<Properties = Record<string, unknown>>(
           delete properties[hookName as unknown as keyof Properties];
         } else if (sort) {
           delete properties[key];
-          Object.assign(properties, { [key]: value });
+          properties[key as keyof Properties] =
+            value as (typeof properties)[keyof Properties];
         }
       }
       return properties as Properties;
     }
 
     function css(
-      ...properties: [
+      ...styles: [
         WithHooks<HookProperties, Properties>,
         ...(WithHooks<HookProperties, Properties> | undefined)[],
       ]
     ): Properties {
-      const styles = [{}].concat(
-        JSON.parse(JSON.stringify(properties)) as typeof properties,
-      ) as typeof properties;
       do {
         const current = styles[0];
         const next = styles[1] || ({} as WithHooks<HookProperties, Properties>);
-        if (options?.sort) {
-          for (const k in next) {
+        for (const k in next) {
+          if (options?.sort) {
```

Comment:
I'm curious to know more about this. I would've guessed it would be better to nest the `if` statement inside of the `for` loop, based on the following:
1. Fewer lookups (the `options.sort` option)
2. The `options?.sort` condition being tested only once.

Reply:
I removed the Object.assign entirely, so instead of only "clearing" current and then merging both current and next, I decided to take current as the base and merge all styles in at once. Therefore, I do both: clear and merge in one loop. That's why the if is inside 🙃